creator/studies/migrations/0031_patch_dataservice.py

```diff
@@ -0,0 +1,118 @@
+# Generated by Django 2.2.24 on 2021-11-02 18:23
+
+"""
+This migration patches the new attributes into the DataService Study
+entities using the spreadsheet found in this migrations folder.
+"""
+
+from django.conf import settings
+from django.db import migrations, models
+
+import pandas as pd
+import os
+import requests
+
+
+def patch_helper(kf_id, study_dict):
+    """
+    Helper function to patch the data in study_dict to the kf_id Study in
+    DataService.
+    """
+    # Patch DataService
+    try:
+        resp = requests.patch(
+            f"{settings.DATASERVICE_URL}/studies/{kf_id}",
+            json=study_dict,
+            timeout=settings.REQUESTS_TIMEOUT,
+            headers=settings.REQUESTS_HEADERS,
+        )
+    except requests.exceptions.RequestException as e:
+        raise Exception(f"Problem updating study {kf_id}: {e}")
+    
+    # Raise an error if it looks like study failed to update
+    if not resp.status_code == 200 or "results" not in resp.json():
+        error = resp.json()
+        if "_status" in error:
+            error = error["_status"]
+        if "message" in error:
+            error = error["message"]
+        raise Exception(f"Problem updating study {kf_id}: {error}")
+
+
+def patch_attrs(apps, schema_editor):
+    """
+    Patch the short_code, domain, and program attributes of Study objects
+    in the DataService using the provided spreadsheet.
+    """
+    # Do nothing if update DataService feature is not enabled
+    if not (
+        settings.FEAT_DATASERVICE_UPDATE_STUDIES
+        and settings.DATASERVICE_URL
+    ):
+        return
+
+    df = pd.read_csv(
+        os.path.abspath("creator/studies/migrations/study_metadata.csv")
+    )
+    short_codes = pd.Series(df.code.values, index=df.kf_id).to_dict()
+    domains = pd.Series(df.domain.values, index=df.kf_id).to_dict()
+    programs = pd.Series(df.program.values, index=df.kf_id).to_dict()
+
+    Study = apps.get_model("studies", "Study")
+    mappings = {
+        "Birth Defect": "BIRTHDEFECT",
+        "Other": "OTHER",
+        "Cancer": "CANCER",
+        "Cancer,Birth Defect": "CANCERBIRTHDEFECT",
+        "COVID-19": "COVID19",
+    }
+
+    for study in Study.objects.all():
```

Comment:
Since this migration runs last, couldn't you just pull the values you need for short_code, domain, program from the study instance in study creator?